packages/components/src/Modal/ConfirmationModal/_docs/ConfirmationModal.stories.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { useState } from "react"
+import { Meta, StoryObj } from "@storybook/react"
+import isChromatic from "chromatic"
+import { Text } from "~components/Text"
+import { ConfirmationModal, ConfirmationModalProps } from "../index"
+
+const IS_CHROMATIC = isChromatic()
+
+const ConfirmationModalWithState = (
+  args: ConfirmationModalProps
+): JSX.Element => {
+  const [isOpen, setIsOpen] = useState(IS_CHROMATIC)
+
+  const handleOpen = (): void => setIsOpen(true)
+  const handleClose = (): void => setIsOpen(false)
+
+  return (
+    <>
+      <button type="button" onClick={handleOpen}>
+        Open Modal
+      </button>
+      <ConfirmationModal
+        {...args}
+        isOpen={isOpen}
+        onConfirm={handleClose}
+        onDismiss={handleClose}
+      />
+    </>
+  )
+}
```

Comment:
If you make this into a template Story (have this as the `render`, then spread the whole Story into the other Stories), you won't need to type `args` nor the return type, and it will have intellisense (based on the meta) that tells you if you're missing args.